dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -32,141 +31,89 @@ namespace PushNotificationHelpers
     }
 
     winrt::Windows::DateTime PushNotificationChannel::ExpirationTime()
     {
-        if (m_channel)
-        {
-            return m_channel.ExpirationTime();
-        }
-        else
-        {
-            return m_channelInfo.channelExpiryTime;
-        }
+        return m_channelInfo.channelExpiryTime;
     }
 
     void PushNotificationChannel::Close()
     {
         try
         {
-            if (m_channel)
-            {
-                m_channel.Close();
-            }
-            else
-            {
-                THROW_IF_FAILED(PushNotifications_CloseChannel(m_channelInfo.appUserModelId.c_str(), m_channelInfo.channelId.c_str()));
-            }
-
+            THROW_IF_FAILED(PushNotifications_CloseChannel(m_channelInfo.appId.c_str(), m_channelInfo.channelId.c_str()));
             PushNotificationTelemetry::ChannelClosedByApi(S_OK);
         }
 
         catch (...)
         {
-            auto channelCloseException = hresult_error(to_hresult());
+            auto channelCloseException { hresult_error(to_hresult()) };
 
             PushNotificationTelemetry::ChannelClosedByApi(channelCloseException.code());
 
             if (channelCloseException.code() != HRESULT_FROM_WIN32(ERROR_NOT_FOUND))
             {
-                throw hresult_error(to_hresult());
+                throw channelCloseException;
             }
         }
     }
 
     winrt::event_token PushNotificationChannel::PushReceived(winrt::Windows::TypedEventHandler<winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel, winrt::Microsoft::Windows::PushNotifications::PushNotificationReceivedEventArgs> handler)
     {
-        if (PushNotificationHelpers::IsPackagedAppScenario())
+        bool registeredEvent { false };
         {
-            if (m_channel)
-            {
-                return m_channel.PushNotificationReceived([weak_self = get_weak(), handler](auto&&, auto&& args)
-                {
-                    if (auto strong = weak_self.get())
-                    {
-                        auto pushArgs = winrt::make<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationReceivedEventArgs>(args);
-                        pushArgs.Handled(true);
-                        handler(*strong, pushArgs);
-                    };
-                });
-            }
-            else
-            {
-                // The channelUri is directly obtained when we request Channel from UDK using RemoteId
-                auto lock = m_lock.lock_exclusive();
-
-                if (!m_foregroundHandlerCount)
-                {
-                    auto appUserModelId{ winrt::Microsoft::Helpers::GetAppUserModelId() };
-
-                    THROW_IF_FAILED(PushNotifications_RegisterNotificationSinkForFullTrustApplication(appUserModelId.get(), this));
-                }
+            auto lock { m_lock.lock_shared() };
+            registeredEvent = bool(m_foregroundHandlers);
```

Comment:
I don't understand why we need this. Isn't this always guaranteed to be true? So why bother?

Reply:
You are just complicating the logic with this check.

Reply:
the bool operator checks if there are any registered handlers in the winrt::event. Removes the need for a handler count.

https://docs.microsoft.com/en-us/uwp/cpp-ref-for-winrt/event#eventoperator-bool

Reply:
oh thats cool that they overloaded the bool operator this way. Would have expected an isEmpty or something similar.